cirq/study/sweeps.py

```diff
@@ -101,6 +101,28 @@ def __iter__(self) -> Iterator[resolver.ParamResolver]:
         for params in self.param_tuples():
             yield resolver.ParamResolver(collections.OrderedDict(params))
 
+    def __getitem__(self, val) -> Union[Params, Iterator[Params]]:
+        single_item = False
+        if isinstance(val, int):
+            val = slice(val, val + 1, None)
+            single_item = True
+        if not isinstance(val, slice):
+            raise TypeError(
+                'Sweep indices must be either int or slices, not {}'\
+                    .format(type(val)))
+        if (val.start is not None and val.start < 0) or \
+           (val.stop is not None and val.stop < 0) or \
+           (val.step is not None and val.step < 0):
+            raise ValueError(
+                'Cannot slice/access sweeps using negative indices.')
```

Comment:
I think we would want to support as much of the standard python slicing syntax as we can. Generic iterables do not have a known length, so `itertools.islice` can't support negative indexes, but sweeps _do_ have know length so we should be able to handle these.

Reply:
I've updated the code to support this, let me know what you think now.